wattsonic/integrationpluginwattsonic.cpp

```diff
@@ -255,8 +249,13 @@ void IntegrationPluginWattsonic::setupWattsonicConnection(ThingSetupInfo *info)
             battery->setStateValue(batteryBatteryLevelStateTypeId, connection->SOC() / 100.0);
             battery->setStateValue(batteryBatteryCriticalStateTypeId, connection->SOC() < 500);
         }
-
     });
+}
 
-
+void IntegrationPluginWattsonic::setConnectedStates(Thing *thing, bool connected)
+{
+    thing->setStateValue("connected", connected);
+    foreach (Thing *child, myThings().filterByParentId(thing->id())) {
+        child->setStateValue("connected", connected);
```

Comment:
have no way of testing it, but seems like in case of disconnection you may want to set currentPower etc to 0 like other plugins do?